lib/SPIRV/libSPIRV/SPIRVAsm.h

```diff
@@ -0,0 +1,155 @@
+//===- SPIRVAsm.h -                                           --*- C++ -*-===//
+//
+//                     The LLVM/SPIRV Translator
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+/// \file
+///
+/// This file defines the inline assembler entries defined in SPIRV spec with op
+/// codes.
+///
+//===----------------------------------------------------------------------===//
+
+#ifndef SPIRV_LIBSPIRV_SPIRVASM_H
+#define SPIRV_LIBSPIRV_SPIRVASM_H
+
+#include "SPIRVEntry.h"
+#include "SPIRVInstruction.h"
+#include "SPIRVStream.h"
+
+#include <cassert>
+#include <map>
+#include <tuple>
+#include <vector>
+
+namespace SPIRV {
+
+class SPIRVAsmTargetINTEL : public SPIRVEntry {
+public:
+  static const SPIRVWord FixedWC = 2;
```

Comment:
According to the spec `OpAsmTargetINTEL` must have 3 words

Reply:
FixedWC is not a total word count. There are also could be more than 3 words.

Reply:
Right. There must be at least 3 words, because `Asm Target` literal is not optional and it takes at least one word.

Reply:
Do you suggest subtract one from WC in constructor?
`: SPIRVEntry(M, FixedWC + getSizeInWords(TheTarget) - 1, OC, TheId),`

Reply:
No, I think it is ok as is.